src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -2038,13 +2058,80 @@ MulticopterPositionControl::task_main()
 				}
 			}
 
-			/* control roll and pitch directly if no aiding velocity controller is active */
 			if (!_control_mode.flag_control_velocity_enabled) {
 				_att_sp.roll_body = _manual.y * _params.man_roll_max;
 				_att_sp.pitch_body = -_manual.x * _params.man_pitch_max;
 
-				/* only if optimal recovery is not used, modify roll/pitch */
-				if (_params.opt_recover <= 0) {
+				/* If ACRO attitude control mode */
+				if (_vehicle_status.nav_state == vehicle_status_s::NAVIGATION_STATE_ACRO) {
```

Comment:
This PR creates a new implementation of ACRO mode, which controls both rates and attitude. It would become much simpler if it simply replaced the rates-only acro mode. Perhaps that would be better. This version of ACRO differs from STABILIZE only in the way that RC inputs are interpreted: roll/pitch inputs are interpreted as rate demands, instead of angle setpoints.